### ISSUE TITLE
[doc] 5.3.1  Disabling Automatic Updates - document manual steps if automation is deactivated

### ISSUE THREAD

**Issue by Martin-Weiss** [SUSE/doc-caasp#652]
5.3.1  Disabling Automatic Updates

https://susedoc.github.io/doc-caasp/master/caasp-admin/single-html/#_disabling_automatic_updates
----

What does a customer have to do to update his cluster "correct" in case he turns of automatic updates following this way? I assume we should document this in addition to the "turn off" function...

**Comment by r0ckarong** [comment]
@ereslibre @MaximilianMeister

**Comment by Martin-Weiss** [comment]
I assume a manual execution of "skuba-update" on all nodes makes sense for manuall steps and let kured reboot..

But - Just realized that we also need a more detailed explanation on how kured works for rebooting the cluster. It seems that a check "reboot needed?" happens every hour - but if during this time an other node has the "lock" the next reboot check and reboot try will be an hour later.. In a large cluster this means the reboot process for the whole cluster can take ages if I understand this right..

**Comment by ereslibre** [comment]
> But - Just realized that we also need a more detailed explanation on how kured works for rebooting the cluster. It seems that a check "reboot needed?" happens every hour - but if during this time an other node has the "lock" the next reboot check and reboot try will be an hour later.. In a large cluster this means the reboot process for the whole cluster can take ages if I understand this right..

@MaximilianMeister, do you have more insight on this? If Max doesn't know I will have a look at Kured's code to confirm.

**Comment by MaximilianMeister** [comment]
>  do you have more insight on this? If Max doesn't know I will have a look at Kured's code to confirm.

@Martin-Weiss the kured lock is global and not per node (in contrast to a manual `skuba-update`). Shall we make that more clear in the docs? i hope this answers your doubts

**Comment by Martin-Weiss** [comment]
@MaximilianMeister - I understand that the lock is global - but is held by one node. And each node checks the lock only once per hours if I understand it right - at the same time when it checks for "reboot needed".

So if we boot up the whole cluster at more or less the same point in time it is pretty likely that we have one node reboot per hour - so a 10 node cluster needs up to ~11 hours to get rebooted completely. With daily "reboot needed" patches of a >24 node cluster - the process would never finish and nodes would also have different patchlevels.

In case the customer has a scheduled reboot window of two hours on a Saturday night - this is getting tricky ;-).

I am not sure why the default check is only every 60 Minutes. Probably we need two checks - one every 60 Minutes for "reboot needed/required" and after this an other poll every Minute against the lock?

And yes - some more detailed documentation for the operators would be great! Thanks!

**Comment by MaximilianMeister** [comment]
> And each node checks the lock only once per hours if I understand it right - at the same time when it checks for "reboot needed".

yeah also it checks the lock only if a reboot is required

> I am not sure why the default check is only every 60 Minutes. Probably we need two checks - one every 60 Minutes for "reboot needed/required" and after this an other poll every Minute against the lock?

I agree this should be separated somehow, the check is only [one for all and always the same](https://github.com/weaveworks/kured/blob/master/cmd/kured/main.go#L310) every hour in our case. Reducing the default time window of the check should already help in this case. However here in this context of the issue we're more talking about a manual reboot procedure by the user, which also means that kured is disabled.

So for this issue here we could add an example case of a manual update of a X node cluster to the documentation.

For the kured check frequency, and possibly enhancements, which have to be done upstream, we should open another issue i'd say. Even if it's just about changing the default 60min to something shorter like 5min.

What do you think?

**Comment by Martin-Weiss** [comment]
> 
> 
> > And each node checks the lock only once per hours if I understand it right - at the same time when it checks for "reboot needed".
> 
> yeah also it checks the lock only if a reboot is required
> 
> > I am not sure why the default check is only every 60 Minutes. Probably we need two checks - one every 60 Minutes for "reboot needed/required" and after this an other poll every Minute against the lock?
> 
> I agree this should be separated somehow, the check is only [one for all and always the same](https://github.com/weaveworks/kured/blob/master/cmd/kured/main.go#L310) every hour in our case. 

But it depends on the time when the kured POD started - so if not all nodes booted at the same time / started the POD at the same time - the check and reboot happens at different times within an hour.

>Reducing the default time window of the check should already help in this case. However here in this context of the issue we're more talking about a manual reboot procedure by the user, which also means that kured is disabled.

Yes and no - disabling the automated updates first means "set --annotate-only" in /etc/sysconfig/skuba-update and with that manual execution of skuba-update at the time the customer wants that.

This might then create the /var/run/reboot-needed and/or /var/run/reboot-required.

But then the reboot process of the cluster via kured kicks in which needs to be configured properly or needs to be disabled and then the manual "drain, reboot, uncordon" needs to be done.

IMO we have to cover different combinations of "patch installation on/off/scheduled" and "kured on/off/schedule configured".

> 
> So for this issue here we could add an example case of a manual update of a X node cluster to the documentation.

We have to be clear that never more than two patchlevels must exist in a given cluster..
(the current running and the one currently in roll-out)

> 
> For the kured check frequency, and possibly enhancements, which have to be done upstream, we should open another issue i'd say. Even if it's just about changing the default 60min to something shorter like 5min.
> 
> What do you think?

First we have to document how to operate the currently given options (until we get the feature that allows the kured schedule configuration). I believe I have created an issue for the kured schedule configuration already.

Upstream we should ask why they have decided for the combined check once per hour and if it would make sense to add a second timer - check every hour for "reboot needed" but if "reboot needed is found" check the lock every 5 Minutes or so..
